src/shared/locked-status/use-check-lock-status.js

```diff
@@ -10,96 +12,161 @@ import { useDataValueSet } from '../use-data-value-set/use-data-value-set.js'
+    if (!selectedPeriod) {
+        return
+    }
+
+    const dataSet = selectors.getDataSetById(metadata, dataSetId)
+    if (!dataSet) {
+        return
+    }
+
+    const { expiryDays, dataInputPeriods } = dataSet
+    if (dataInputPeriods.length === 0 && expiryDays === 0) {
+        // Nothing to check here then
+        return
+    }
+
+    let clientLockDate
+    const currentDate = fromClientDate(getCurrentDate())
+
+    if (dataInputPeriods.length > 0) {
+        const applicableDataInputPeriod =
+            selectors.getApplicableDataInputPeriod(
+                metadata,
+                dataSetId,
+                selectedPeriod.id
+            )
+
+        if (!applicableDataInputPeriod) {
+            // If there are defined data input periods, but there is not one
+            // for this selected period, then the form is closed
+            return {
+                state: LockedStates.LOCKED_DATA_INPUT_PERIOD,
+                lockDate: null,
+            }
+        }
+
+        const { openingDate, closingDate } = applicableDataInputPeriod
+        // openingDate and closingDate can be undefined.
+        // They are local to the server
+        const parsedOpeningDate =
+            openingDate && fromServerDate(new Date(openingDate))
+        const parsedClosingDate =
+            closingDate && fromServerDate(new Date(closingDate))
+
+        if (
+            (openingDate &&
+                currentDate.serverDate < parsedOpeningDate.serverDate) ||
+            (closingDate &&
+                currentDate.serverDate > parsedClosingDate.serverDate)
+        ) {
+            return {
+                state: LockedStates.LOCKED_DATA_INPUT_PERIOD,
+                lockDate: null,
+            }
+        }
 
-    if (!dataInputPeriod) {
-        return false
+        // If we're here, the form isn't (yet) locked by the data input period.
+        // Set the clientLockDate to the input period's closing date (if this
+        // input period has one) before also checking expiry days.
+        // This might still be undefined, but that's okay
+        clientLockDate = parsedClosingDate?.clientDate
     }
 
-    const currentDateAtServerTimeZone = new Date(currentDayString)
-    const openingDate = new Date(dataInputPeriod.openingDate)
-    const closingDate = new Date(dataInputPeriod.closingDate)
+    if (expiryDays > 0) {
+        // selectedPeriod.endDate is a string like '2023-02-20' --
+        // convert that to a date and get time to get ms to add expiry days.
+        // Also add one day more because selectedPeriod.endDate is the START
+        // of the period's last day (00:00), and we want the end of that day
+        // (confirmed with backend behavior).
+        // Converting the selectedPeriod.endDate string to an object creates
+        // it in the client's timezone; parse it from there.
+        const expiryDate = fromClientDate(
+            new Date(
+                new Date(selectedPeriod.endDate).getTime() +
```

Comment:
Aha, thanks for pointing this out -- I'll look into this 🤔

Reply:
Ah I think I get it -- I guess this was accidentally working because server was in UTC time, so the `new Date('2023-02-23')` happened to give the right client date. I pushed a commit to parse the date from the correct server date instead, like in other places in the code here